doc/rtd/spelling.py

```diff
@@ -0,0 +1,79 @@
+import pathlib
+import re
+
+import enchant
+
+
+class WordListFilter(enchant.tokenize.Filter):
+    word_list = "spelling_word_list.txt"
+    regex_list = "spelling_regex_list.txt"
+
+    def __init__(self, *args, **kwargs):
+        """Use two files for ignoring correctly spelled words
+
+        - spelling_word_list.txt: a list of exact matches to ignore
+        - spelling_regex_list.txt: a list of regular expressions to ignore
+
+        Splits tokens on "/" and "-".
+        """
+        super().__init__(*args, *kwargs)
+        directory = pathlib.Path(__file__).parent
+        with open(directory.joinpath(self.word_list)) as f:
+            lines = f.read().splitlines()
+            self._validate_lines(lines)
+            self.word_set = set(lines)
+        with open(directory.joinpath(self.regex_list)) as f:
+            self.regex_set = set(f.read().splitlines())
+        print(f"Loaded {self.word_list}: {self.word_set}")
+        print(f"Loaded {self.regex_list}: {self.regex_set}")
+
+    def _validate_lines(self, lines):
+        """Assert that the word_list file is legible and orderly"""
+        if lines != sorted(lines):
+            first_missordered = next_item = previous_item = None
+            for item_a, item_b in zip(lines, sorted(lines)):
+                if first_missordered:
+                    next_item = item_a
+                    break
+                elif item_a != item_b:
+                    first_missordered = item_a
+                else:
+                    previous_item = item_a
+            unordered = (
+                f"[..., {previous_item}, {first_missordered}, "
+                f"{next_item}, ...]"
+            )
+            raise Exception(
+                f"Unsorted {self.word_list} detected. "
+                f"Please sort for legibility. Unordered list: {unordered}"
+            )
+        for line in lines:
```

Comment:
unless it's the first or last word, we're never going to get here because an uppercase letter will sort differently and get triggered by the unsorted detection above (but print an unhelpful `unordered` list).

Also, if you have the full text, something like `if text != text.lower()` would be a simpler check.

Reply:
> we're never going to get here because an uppercase letter will sort differently and get triggered by the unsorted detection above (but print an unhelpful unordered list).

any word that begins with a lower case letter breaks this assumption, and tech has more than a couple of those (camelCase, iPod etc)

Reply:
Ahhh, I see. Yes good point, ordered first would be better.